### DISCUSSION TITLE
Theme Changes Won't Stick

### DISCUSSION THREAD

**Discussion by steveveach** [EdgeTX/edgetx#2388] [Q&A]
Hi everyone,

I've been using the Greenfields theme on my 2.6.0 RadioMaster TX16S Mark II Max running EdgeTX 2.6.0 for weeks with no problems. Then suddenly yesterday, after I powered it up, the default theme was back as the active theme. Now when I switch it back to the Greenfields theme, it won't save that theme as active and always reverts back to the default. As a matter of fact, now _any_ theme changes I make will revert back to the default. I've swapped the SD card but that didn't help.

The only thing I've done differently lately is started to use the EdgeTX Companion but I can't imagine why that would have caused this odd behavior. I am new to EdgeTX but this has me baffled. Maybe it's user error somehow.

Thanks!

**Comment by pfeerick** [comment]
The fact it suddenly changed, and now won't change back would typically make me think it's an SD card issue, but it seems like you've already checked for that. It hasn't done any sudden "long" startups (which indicates an SD issue which there was a recent bugfix for)?

If you change the theme on the radio, and power cycle it, and it doesn't stick, it's unlikely that Companion will be at fault, but stranger things have happened. But as long as you're using the same version of Companion you're probably ok there. Hm... just in case there was a bug around saving the theme setting - try changing your theme, and then go edit something simple in the radio settings - I dunno... change the timezone setting as that doesn't matter anyway unless you're using a GPS. That will trigger the radio to save settings, just in case changing the theme isn't actually being saved.

**Comment by steveveach** [comment]
No long startup issues either so not likely the SD card. I'll try your idea later today. Thanks!

**Comment by steveveach** [comment]
**pfeerick,** I had a chance to try this. Very odd results. With more careful observation, this is what I'm now seeing:

- Immediately after making the change to Greenfields, I see the correct theme in all screens as expected.
- After power cycling, the theme becomes a mix of Default elements and Greenfields elements!

I have no idea how this is happening. Pilot error? A bug? Hard to believe it's a bug since this is ver. 2.6.0 and has been out for a while given we're now up to stable ver. 2.7.1.

Thanks.

![PXL_20220929_170343300 MP~2](https://user-images.githubusercontent.com/48031888/193095096-9fff2e9e-3481-4c9d-8637-ff51055b59e5.jpg)

**Reply by steveveach** [reply]
Update: I just deleted all files in the SD card THEME folder, then copied all original theme files from a 2.6.0 backup folder on my computer to the SD card. Of course, when I booted up, the Default theme was active. I then changed the theme to the Greenfields theme. Once I power cycled, I was back to the mixed theme you see above! I am truly baffled at this point.

**Reply by steveveach** [reply]
I just changed the theme to Handpainted Sky and was surprised to find it appears to be working perfectly. No odd mix of Default and Handpainted Sky elements in the display. I power cycled twice and it's still fine. Very, very odd.
